### PR TITLE
Update the sdk build to target net8.0

### PR DESCRIPTION
**Pull request by marcpopMSFT**

### CODE REVIEW

**Comment by marcpopMSFT** [comment]
@dougbu do you happen to know how to avoid the aspnet analyzer exceptions?  This is our first try to move the sdk to targeting 8.0 and using the 8.0 sdk.
CSC : error AD0001: Analyzer 'Microsoft.AspNetCore.Analyzers.Http.RequestDelegateReturnTypeAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [D:\a\1\s\src\Tests\Microsoft.AspNetCore.Watch.BrowserRefresh.Tests\Microsoft.AspNetCore.Watch.BrowserRefresh.Tests.csproj]
##[error]CSC(0,0): error AD0001: (NETCORE_ENGINEERING_TELEMETRY=Build) Analyzer 'Microsoft.AspNetCore.Analyzers.Http.RequestDelegateReturnTypeAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.

**Comment by dougbu** [comment]
@marcpopMSFT I think @JamesNK and @captainsafia are our HTTP analyzer specialists. See a few commits from @halter73 too. Hope one of them can help you.

**Comment by JamesNK** [comment]
I’ll take a look.

**Comment by JamesNK** [comment]
I've identified the problem. The fix is in aspnetcore so you'll need to wait for the update to flow through.

**Comment by marcpopMSFT** [comment]
@JamesNK the SDK main branch is on aspnetcore [8.0.0-alpha.1.22517.3](https://github.com/dotnet/aspnetcore/compare/e35f2ea...fa2fd21) and I tried rerunning this PR but it still failed with the same error. What else needs to be done to unblock this?

**Comment by JamesNK** [comment]
When I ran build on main I got 8.0.0-alpha.1.22512.9 of the SDK restored. Its commit was from before my changes - https://github.com/dotnet/aspnetcore/commit/361e62888c49aca87f6be231e1442ee49a46a042

**Comment by kasperk81** [comment]
do you get the old version with https://github.com/dotnet/sdk/pull/28625 as well? if not, after merging that pr try merging main to this pr's branch

**Comment by kasperk81** [comment]
https://github.com/dotnet/aspnetcore/pull/44534#discussion_r997832717

**Comment by JamesNK** [comment]
> When I ran build on main I got 8.0.0-alpha.1.22512.9 of the SDK restored. Its commit was from before my changes - [dotnet/aspnetcore@361e628](https://github.com/dotnet/aspnetcore/commit/361e62888c49aca87f6be231e1442ee49a46a042)

You're still using the same SDK without the fix...

image

File: global.json
Comment (JamesNK) [inline global.json:3]: It passes on my machine when I update the SDK:
```suggestion
    "dotnet": "8.0.100-alpha.1.22514.14",
```

Comment (marcpopMSFT) [reply global.json]: Thanks. Sounds like it's a difference between the SDK built and the SDK used for building (we call it the stage 0 SDK). Let's try updating that.

**Comment by kasperk81** [comment]
next error is about updating https://github.com/dotnet/sdk/blob/4e8e433c00d3cb07ba67a300c40af66a82ec36bb/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets#L18 to net8.0

**Comment by nagilson** [comment]
Seems this is still looking for `net7.0`:
```bash
    C:\h\w\B756097F\p\ex\msbuildExtensions\Microsoft\Microsoft.NET.Build.Extensions\Microsoft.NET.Build.Extensions.NETFramework.targets(114,5): error MSB4062: The "AddFacadesToReferences" task could not be loaded from the assembly 

C:\h\w\B756097F\p\ex\msbuildExtensions\Microsoft\Microsoft.NET.Build.Extensions\\tools\net7.0\Microsoft.NET.Build.Extensions.Tasks.dll.
 Could not load file or assembly 'C:\h\w\B756097F\p\ex\msbuildExtensions\Microsoft\Microsoft.NET.Build.Extensions\tools\net7.0\Microsoft.NET.Build.Extensions.Tasks.dll'. 
 The system cannot find the path specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\h\w\B756097F\t\dotnetSdkTests\r1lisw1i.eln\Anet462Projec---F15591BC\TestApp\TestApp.csproj]
```

**Comment by marcpopMSFT** [comment]
@lewing GetPackDefinitionLocations is failing because it sees Microsoft.NETCore.App.Runtime.osx-arm64 and osx.x64 in both the 8 and 7 manifests from what I can tell. CC @dsplaisted 
image

**Comment by nagilson** [comment]
@dotnet/aspnet-blazor-eng Looks like a lot of the remaining failures (or all) are tests in blazor-wasm. Do these need to be updated to allow targeting 8.0? May you please take a look? cc @marcpopMSFT

**Comment by SteveSandersonMS** [comment]
@marcpopMSFT I added a couple of commits that update more TFM references, since that's what it took to make the Blazor WebAssembly tests pass locally for me in this branch. Now the CI results are reporting different errors unrelated to Blazor WebAssembly, and looking at their failure logs, it seems that some other components are still referencing `net7.0` so it looks like there are more TFM references to update elsewhere.

**Comment by marcpopMSFT** [comment]
I can poke at some of the sdk test failures to see if there are more net7 references but there are still a lot of razor/blazor tests that appear failing for baseline mismatches unless I'm misreading the results.

**Comment by TanayParikh** [comment]
> a lot of razor/blazor tests that appear failing for baseline mismatches

Looks like this may necessitate baseline regeneration as some assets may've been added or removed. Have https://github.com/dotnet/sdk/pull/27542 currently which I'm hoping to get merged. It'll provide much clearer logging on what the actual mismatch is, as well as how to regenerate the baselines should it be necessary.

**Comment by marcpopMSFT** [comment]
@TanayParikh I'm a bit confused. I ran that script and 23/34 tests pass and the only changes are changing ${Tfm} to net7.0. Is that's what's expected? Should I include that update in my PR?

**Comment by marcpopMSFT** [comment]
Latest change is getting closer.

@dotnet/aspnet-blazor-eng can you take a look at the blazor failures that all appear to be similar to this:
Expected File C:\h\w\9FDD0948\t\dotnetSdkTests\ep3bpqss.izk\Pack_Multiple---7F5CE8C3\bin\Debug\net8.0\PackageLibraryTransitiveDependency.dll to exist, but it does not.

@dotnet/templating-engine-maintainers there are a few template engine tests also failing. Can you take a look as well?

**Comment by marcpopMSFT** [comment]
> @dotnet/aspnet-blazor-eng can you take a look at the blazor failures that all appear to be similar to this: Expected File C:\h\w\9FDD0948\t\dotnetSdkTests\ep3bpqss.izk\Pack_Multiple---7F5CE8C3\bin\Debug\net8.0\PackageLibraryTransitiveDependency.dll to exist, but it does not.
The tests that are still failing are compiling for net7.0. I don't think the default TFM can be switched though as other tests will fail. Not sure if the set that are failing should be modified now to target net7.0 explicitly.

File: src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.5_0.targets
Comment (RussKie) [inline src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.5_0.targets:31]: We updating this in many places, is it worth changing to this?
```suggestion
    <_BlazorWebAssemblySdkTasksTFM Condition=" '$(MSBuildRuntimeType)' == 'Core'">$(SdkTargetFramework)</_BlazorWebAssemblySdkTasksTFM>
```

**Comment by dreddy-work** [comment]
@dotnet/aspnet-blazor-eng , can anyone take a look here? both WinForms and WPF are blocked from flowing up.

**Comment by vlada-shubina** [comment]
> Latest change is getting closer.
> 
> @dotnet/aspnet-blazor-eng can you take a look at the blazor failures that all appear to be similar to this: Expected File C:\h\w\9FDD0948\t\dotnetSdkTests\ep3bpqss.izk\Pack_Multiple---7F5CE8C3\bin\Debug\net8.0\PackageLibraryTransitiveDependency.dll to exist, but it does not.
> 
> @dotnet/templating-engine-maintainers there are a few template engine tests also failing. Can you take a look as well?

templating tests were fixed in https://github.com/dotnet/sdk/pull/28563/commits/4da0821bba428f83d74db05fed7e4d5ad22a6c46

**Comment by nagilson** [comment]
@dotnet/aspnet-blazor-eng  PTAL, there are at least 10+ PRs blocked by this I can think of now

**Comment by javiercn** [comment]
@nagilson we have PRs out to address this across all the branches, but the one on main is blocked on some other dotnet-watch tests failing. See:
https://github.com/dotnet/sdk/pull/29093

These other PRs are just waiting on confirmation that we can merge on the target branches
https://github.com/dotnet/sdk/pull/29118
https://github.com/dotnet/sdk/pull/29119
https://github.com/dotnet/sdk/pull/29120

**Comment by marcpopMSFT** [comment]
@tmat I believe Noah said that you were already looking into dotnet watch failures. Looks like there are four failures now here in those tests.

**Comment by kasperk81** [comment]
framework error `There is not enough space on the disk.` is unrelated.

merge?

**Comment by nagilson** [comment]
Merge. 😏

**Review by nagilson — APPROVED** [review]

**Pull request merged**
